src/main/java/com/karlofduty/GoldDropRemover/EntityDeathHandler.java

```diff
@@ -3,28 +3,37 @@
 import org.bukkit.Material;
 import org.bukkit.entity.Drowned;
 import org.bukkit.entity.PigZombie;
+import org.bukkit.entity.Piglin;
+import org.bukkit.entity.PiglinBrute;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 import org.bukkit.event.entity.EntityDeathEvent;
 
-public class EntityDeathHandler implements Listener
-{
+public class EntityDeathHandler implements Listener {
     @EventHandler
-    public void onDeath(EntityDeathEvent event)
-    {
-        if(!(event.getEntity() instanceof PigZombie) && !(event.getEntity() instanceof Drowned))
-        {
-            return;
+    public void onDeath(EntityDeathEvent event) {
+        if (event.getEntity() instanceof PigZombie || event.getEntity() instanceof Drowned ||
+            event.getEntity() instanceof Piglin || event.getEntity() instanceof PiglinBrute) {
+            for (int i = 0; i < event.getDrops().size(); i++) {
+                if (isGold(event.getDrops().get(i).getType()))
+                    event.getDrops().remove(i);
+            }
         }
+    }
 
-        for(int i = 0; i < event.getDrops().size(); i++)
-        {
-            if(event.getDrops().get(i).getType().equals(Material.GOLD_INGOT) || event.getDrops().get(i).getType().equals(Material.GOLD_NUGGET) || event.getDrops().get(i).getType().equals(Material.GOLDEN_SWORD))
-            {
-                event.getDrops().remove(i);
-                i--;
```

Comment:
I see you've removed the decrement here, wouldn't that cause the last element to be missed if one was already removed?

Reply:
Fixed, didn't notice that during testing since it's very rare to get multiple gold drops from 1 entity

Reply:
Yeah, its mostly from things like when players got both gold and a gold sword from a zombie pigman in the past